sdk/servicebus/azure-servicebus/azure/servicebus/management/_models.py

```diff
@@ -652,49 +674,98 @@ def _to_internal_entity(self):
 
 
 class SqlRuleFilter(object):
-    def __init__(self, sql_expression=None):
+    """Represents a filter which is a composition of an expression and an action
+    that is executed in the pub/sub pipeline.
+
+    :param sql_expression: The SQL expression. e.g. MyProperty='ABC'
+    :type sql_expression: str
+    :param requires_preprocessing: Value that indicates whether the rule
+     filter requires preprocessing. Default value: True .
+    :type requires_preprocessing: bool
+    """
+    def __init__(self, sql_expression=None, parameters=None, requires_preprocessing=True):
         self.sql_expression = sql_expression
+        self.parameters = parameters
+        self.requires_preprocessing = requires_preprocessing
 
     @classmethod
     def _from_internal_entity(cls, internal_sql_rule_filter):
         sql_rule_filter = cls()
         sql_rule_filter.sql_expression = internal_sql_rule_filter.sql_expression
+        sql_rule_filter.parameters = OrderedDict((kv.key, kv.value) for kv in internal_sql_rule_filter.parameters) \
+            if internal_sql_rule_filter.parameters else OrderedDict()
```

Comment:
if not internal parameter, return empty dict or None?
for symmetry, I see the logic in `_to_internal_entity` is `if self.parameters else None`
```suggestion
        sql_rule_filter.parameters = OrderedDict((kv.key, kv.value) for kv in internal_sql_rule_filter.parameters) \
            if internal_sql_rule_filter.parameters else None
```

Reply:
Initializing the parameters allows the users to write simpler code. They don't have to use "if" to test whether it's None.

Reply:
I'll take your point on this.